apm/src/app/movies/movies.service.ts

```diff
@@ -0,0 +1,30 @@
+import { HttpClient, HttpErrorResponse } from '@angular/common/http';
+import { Injectable } from '@angular/core';
+import { catchError, Observable, tap, throwError } from 'rxjs';
+import { IMoviesResult } from './movies';
+
+@Injectable()
+export class MoviesService {
+
+  constructor(private readonly http: HttpClient) { }
+
+  public getMovies(): Observable<IMoviesResult>  {
+    const url = "https://swapi.dev/api/films/";
```

Comment:
This could be moved out to be a static readonly variable on the MoviesService class, so that it is not created everytime there is a call to this method